### DISCUSSION TITLE
Simple asset copying

### DISCUSSION THREAD

**Discussion by Theramar** [xmake-io/xmake#4160] [Q&A]
### Is your feature request related to a problem? Please describe.

I have an `asset/` folder which contains an `img.png` file.
The executable needs to read the file therefore `img.png` must be copied to the executable dir, now problems arise:
- how to query the executable dir
- how to cache `img.png` to prevent copying every time a new build starts

### Describe the solution you'd like

> "[...] the configuration syntax is much concise and intuitive. As such, it's friendly to novices..."

source: [xmake introduction](https://xmake.io/#/about/introduction)
 ## Proposal
``` lua
target:add_assets("asset/img.png")
target:set_assets("asset/img.png")
```
`add_assets()` should:
- copy the files in the executable dir when installing and packaging
- add a symlink to the files in the executable dir when building
  - reason: a build is not an install: build is used for development. A development environment obviously contains the asset files therefore an expensive copy is not necessary 

## utils.bin2c (i.e. convert `img.png` to `img.png.h`)
- increases code size therefore decreases performance
- slow compile time for large files
- does not support file writing
- obsolete: C23 supports `#embed`
- it is not always feasible to hardcode the path,  i.e. 
  ``` cpp 
  #include "img.png.h" 
  ```
  sometimes users compute the path at run-time or at least at compile-time with `constexpr` not with preprocessor black magic:
  ``` cpp 
  constexpr auto paths = std::to_array<const char*>({"img", "img2"});
  constexpr auto pathExtension = ".png";
  constexpr decltype(paths) getPaths() {} // adds pathExtension at the end of every path
  ```

### Additional context

- xmake automatically puts my executable at `/build/windows/x64/debug/`
- same problem with `utils.glsl2spv`: the executable needs to read shaders

**Comment by waruqi** [comment]
```lua
target("test")
     add_files("src/*.cpp")
     set_configdir("$(buildir)/$(plat)/$(arch)/$(mode)")
     add_configfiles("assets/*.png", {onlycopy = true})
```

**Comment by Theramar** [comment]
thank you!

[a comment by an automated account is not shown]
